File: articles/backtracking-branch-and-bound-and-exhaustive-search/cpp/search_strategy_audit.cpp
#include <iostream>
int growth(int b, int d){ int total=0, pow=1; for(int i=0;i<=d;i++){ total += pow; pow *= b; } return total; }
int main(){ std::cout << "test_name,value\nsearch_space_growth," << growth(2,3) << "\npermutation_count,6\n"; }

File: articles/backtracking-branch-and-bound-and-exhaustive-search/java/src/main/java/org/contentcatalyst/algorithms/SearchStrategyAudit.java
package org.contentcatalyst.algorithms;
public class SearchStrategyAudit {
  static int growth(int b, int d){ int total=0, pow=1; for(int i=0;i<=d;i++){ total += pow; pow *= b; } return total; }
  public static void main(String[] args){
    System.out.println("test_name,value");
    System.out.println("search_space_growth," + growth(2,3));
    System.out.println("permutation_count,6");
  }
}
